site/components/Provider/Provider.tsx

```diff
@@ -37,7 +41,11 @@ const { wallets } = getDefaultWallets();
 const config = getDefaultConfig({
   appName: 'rainbowkit.com',
   projectId,
-  chains: [mainnet, polygon, optimism, arbitrum, base, zora, bsc],
+  // TODO: Remove `as` type assertion operator
+  chains: [mainnet, polygon, optimism, arbitrum, base, zora, bsc] as [
+    Chain,
+    ...Chain[],
+  ],
```

Comment:
Marked this as a `TODO` since it's not big issue and won't cause crashes. `viem` just throws some type errors for chains. 

At least the latest `wagmi` version fixes some crucial bugs with hydration and server-side rendering. Hopefully it's worth merging this PR as is.

Reply:
@magiziz Was able to resolve this by using strict typing in the `site/` project and fixing a few misaligned types there.